From 30f1fdb37db44ad5b011b6be3a1f138d7cd57647 Mon Sep 17 00:00:00 2001
Message-Id: <30f1fdb37db44ad5b011b6be3a1f138d7cd57647.1440106296.git.chang-joon.lee@intel.com>
In-Reply-To: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
References: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Fri, 7 Aug 2015 14:49:52 +0530
Subject: [PATCH 10/11] SQUASHME! [VPG]: drm/i915: dont write to pfit reg when
 pfit disabled

When panel fitter is disabled and if there is and change in the mode
or the pipe select bit of pfit_control register dont write to the
register.

SQUASHME: This patch is to be squashed with the below patch
    commit 32ebc82f175e8ddad7a2a2d6aa0b7b13fa9cb15d
    Author: Arun R Murthy <arun.r.murthy@intel.com>
    Date:   Tue Mar 3 14:27:59 2015 +0530

	SQUASHME! [VPG]: drm/i915: Enable panel fitter only on HWC request

Change-Id: I001ebb144d4479aa1f1bbe7710cc6ca1e19d850a
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-42279
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 2efe1c8..5ea7005 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -10943,8 +10943,13 @@ static void i915_commit(struct drm_i915_private *dev_priv,
 	}
 
 	if (dev_priv->pfit_changed) {
-		I915_WRITE(PFIT_CONTROL, intel_crtc->pfit_control);
-		I915_WRITE(PIPESRC(pipe), intel_crtc->scaling_src_size);
+		bool pfit_en = ((intel_crtc->pfit_control & PFIT_ENABLE) ?
+				true: false);
+		if (intel_crtc->pfit_en_status || pfit_en != intel_crtc->pfit_en_status) {
+			I915_WRITE(PFIT_CONTROL, intel_crtc->pfit_control);
+			I915_WRITE(PIPESRC(pipe), intel_crtc->scaling_src_size);
+		}
+		intel_crtc->pfit_en_status = pfit_en;
 		dev_priv->pfit_changed = false;
 	}
 
-- 
1.9.1

